gammapy/maps/core.py

```diff
@@ -1442,6 +1442,27 @@ def from_stack(cls, maps, axis=None, axis_name=None):
             data=np.stack(data), geom=geom.to_cube(axes=[axis]), unit=maps[0].unit
         )
 
+    def split_by_axis(self, axis_name):
```

Comment:
maybe add a `keepdims` argument to allow for axis dropping.

Reply:
Another question, in terms of API, I wonder whether it would not make more sense to generalize `RegionNDMap.iter_by_axis()` to all `Map` types? In that case, the need for the `split_by_axis` is less strong.
Any idea @adonath ?

Reply:
I wanted to add `keepdims`, but it requires changes in `Geom` and `MapAxis`. I can do it in a followup PR. (Maybe it was a bad idea to do the `split_by_axis` in this PR in the first place)

Reply:
I think it is a good suggestion, however there is one detail:

For historical reasons we have `Map.iter_by_image`, which returns the data as array and the corresponding ND- index. I later implemented `RegionNDMap.iter_by_axis()` with the same API and return values. But I think it is not the API the users would like to have. For users one would rather directly return a sliced or indexed `Map` object. The only clean solution I see would be:
- Rename the existing `.iter_by_image` to `.iter_by_image_data` and `.iter_by_axis_data`
- Change `.iter_by_image` and `.iter_by_axis` to directly return `Map` objects
We could still keep `.split_by_axis()`...

Reply:
@AtreyeeS I'll open a PR to implement my suggestion from the comment above. The you can rebase and simplify your implementation...